docs/sql/general/constraints.rst

```diff
@@ -53,6 +53,75 @@ constraint or a table constraint.
 For further details about the meaning of the options see
 :ref:`indices_and_fulltext`.
 
+.. _check_constraint:
+
+``CHECK``
+---------
+
+The CHECK constraint specifies that the value/s of certain column/s must satisfy
+a boolean expression on insert and update.
+
+Syntax:
+
+::
+
+  check(boolean_expression)
+  constraint <name> check(boolean_expression)
+
+With two options:
+
+1) As many check constraints as required, as additional elements of
+   a create table statement.
+   When name is omitted, one is generated with pattern `fqtn_check_uuid`,
+   where:
+
+   - `fqtb`: full qualified table name.
+   - `check`: "check".
+   - `uuid`: 6 random hexadecimal digits (2 octets),
+     e.g. doc_my_table_check_bd3f57640937.
+
+2) A single check constraint is allowed as part of a column definition.
+   When name is omitted, the pattern is `fqtn_column_check_uuid`,
+   where:
+
+   - `column`: column name.
+
+Example:
```

Comment:
I think this is a bit hard to understand. How about:


```suggestion
If `CONSTAINT <name>` is omitted, CrateDB generates a unique name automatically. This name is visible in :ref:`.. ref to information_schema table` and can be used to :ref:`ALTER TABLE DROP` a constraint.
```

Leave out the implementation details of the name generation. (Not that interesting, and allows us to change it later on - as we can treat this as an implementation detail).


```
The CONSTRAINT definition can either be inline with a column, like this::

    cr> CREATE TABLE metrics1 (
    ...     weight REAL CONSTRAINT weight_is_positive CHECK (weight >= 0),
    ... );

Or, also inline, but without explicit name::

    cr> CREATE TABLE metrics2 (
    ...     weight REAL CHECK (weight >= 0),
    ... );


Or on a table level, in which case the `CONSTRAINT` keyword and constraint name is required:


    cr> CREATE TABLE metrics3 (
    ...     weight REAL,
    ...     CONSTRAINT weight_is_positive CHECK (weight >= 0),
    ... );

 ```

Maybe another example which points out that multiple columns can be referenced.